guardian/amigo
# Move wazuh agent parameters to ParameterStore

## What does this change?

The `wazuh-agent` role currently takes 3 input parameters in amigo

 * agent_version
 * manager_ip
 * secret_arn

This becomes painful as more and more recipes use this role, as the parameters needs to be copy pasted around, which also means that if we need to upgrade the agent version, we'll have to make many edits.

This PR re architects this. Those 3 values are now stored in aws ParameterStore

* /stage/stack/amigo-role/wazuh-agent/authentication-secret
* /stage/stack/amigo-role/wazuh-agent/manager-address
* /stage/stack/amigo-role/wazuh-agent/agent-version

With an optional `stage` parameter that can be defined in amigo to override the default value `PROD`. So multiple sets of these 3 parameters might live at one time, which can be useful when testing new agent versions etc

## How to test
Deploy to code and bake `wazuh-agent`. Boot the image in a fresh ec2 instance. Ssh to it and check `journalctl -u wazuh-agent` to see if the agent connected successfully to the manager.

## What then

The recipes that currently use this need to be edited to remove parameters and rebaked/redeployed
